lib/app/features/tokenized_communities/providers/buy_creator_token_notifier_provider.r.dart

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:ion/app/features/tokenized_communities/models/creator_token_buy_request.dart';
+import 'package:ion/app/features/tokenized_communities/providers/providers.r.dart';
+import 'package:ion/app/features/tokenized_communities/views/buy_sell_creator_token_controller.r.dart';
+import 'package:ion/app/features/wallets/utils/crypto_amount_converter.dart';
+import 'package:ion/app/services/logger/logger.dart';
+import 'package:ion_identity_client/ion_identity.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'buy_creator_token_notifier_provider.r.g.dart';
+
+@riverpod
+class BuyCreatorTokenNotifier extends _$BuyCreatorTokenNotifier {
+  @override
+  FutureOr<String?> build(String creatorPubkey) => null;
+
+  Future<void> buy(UserActionSignerNew signer) async {
+    if (state.isLoading) return;
+
+    state = const AsyncValue.loading();
+
+    state = await AsyncValue.guard(() async {
+      final formState = ref.read(buySellCreatorTokenControllerProvider(creatorPubkey));
+
+      final token = formState.selectedPaymentToken;
+      final wallet = formState.targetWallet;
+      final amount = formState.amount;
+
+      if (token == null || wallet == null || amount <= 0) {
+        throw Exception('Invalid form state: token, wallet, or amount is missing');
```

Comment:
Should we display some errors to the end user? If so, we need to add custom exceptions with dedicated error codes (`showErrorModal` / `ref.displayErrors` will handle it)

upd: I see that `ref.displayErrors` is used for this provider, then yea we need custom Exceptions with codes, wdyt?